WeakAurasOptions/WeakAurasOptions.lua

```diff
@@ -1726,6 +1726,148 @@ function OptionsPrivate.OpenTriggerTemplate(data, targetId)
+  frame.scrollList:ReleaseChildren()
+
+  local props = OptionsPrivate.Private.GetAdditionalProperties(data)
+  local sortedProps = {
+    {type = "header", triggerNum = 0, name = "Global Properties"}
+  }
+  for i = 1, 8 do
+    tinsert(sortedProps, {type = "marker", triggerNum = 0, name = "{rt"..i.."}", o = i, n = 8, desc = "|TInterface\\TargetingFrame\\UI-RaidTargetingIcon_"..i..":0|t"})
+  end
+  tAppendAll(sortedProps, BaseDynamicTextCodes)
+  for triggerNum, triggerProps in pairs(props) do
+    if next(props[triggerNum]) then
+      table.insert(sortedProps, {type = "header", triggerNum = triggerNum, name = OptionsPrivate.GetTriggerTitle(data, triggerNum)})
+      for i = 1, 5 do
+        local prop = CopyTable(BaseDynamicTextCodes[i])
+        prop.triggerNum = triggerNum
+        prop.n = 5
+        table.insert(sortedProps, prop)
+      end
+      for name, desc in pairs(triggerProps) do
+        table.insert(sortedProps, {triggerNum = triggerNum, name = name, desc = desc})
+      end
+    end
+  end
+  table.sort(sortedProps, function(a, b)
+    if a.triggerNum ~= b.triggerNum then
+      return a.triggerNum < b.triggerNum
+    elseif a.type ~= b.type then
+      return (a.type or "z") < (b.type or "z")
+    elseif a.type == "marker" and b.type == "marker" then
+      return a.o < b.o
+    elseif a.type == "mini" and b.type == "mini" then
+      return a.o < b.o
+    else
+      return a.name < b.name
+    end
+  end)
+
+  -- Create a modified WeakAurasSnippetButton for each property and add it to ScrollList
+  local lastType, miniGroup
+  for i, prop in ipairs(sortedProps) do
+    if prop.type == "header" then
+      local heading = AceGUI:Create("Heading")
+      heading:SetText(prop.name)
+      heading:SetRelativeWidth(1)
+      heading.label:SetFontObject(GameFontNormalSmall2)
+      frame.scrollList:AddChild(heading)
+    else
+      if ((prop.type == "mini" or prop.type == "marker") and prop.type ~= lastType)
+      then
+        miniGroup = AceGUI:Create("SimpleGroup")
+        miniGroup:SetLayout("Flow")
+        miniGroup:SetAutoAdjustHeight(true)
+        miniGroup:SetRelativeWidth(1)
+        frame.scrollList:AddChild(miniGroup)
+      end
+      local button = AceGUI:Create("WeakAurasSnippetButton")
+      local propIndex = prop.triggerNum > 0 and ("%s"):format(prop.triggerNum) or ""
+      local propPrefix = prop.triggerNum > 0 and ("%%%s."):format(propIndex) or "%"
+      if prop.type == "marker" then
+        button:SetTitle(prop.desc)
+      else
+        button:SetTitle(string.format("|cFFFFCC00%s|r%s", propPrefix, prop.name))
+      end
+      if prop.type == "mini" or prop.type == "marker" then
+        button:SetRelativeWidth((1/prop.n) - 1e-10)
+      else
+        button:SetRelativeWidth(1)
+      end
+      button.title:SetFontObject(GameFontNormal)
+      button.frame:SetHeight(28)
+
+      -- Modify highlight textures hover and push and remove normal texture
```

Comment:
Move this code into a function in WeakAurasSnippetButton. Also you need to consider button reuse by AceGUI, thus all these textures need to be reset to the right value in WeakAurasSnippetButton's OnAcquire method.